tests/test_err.py

```diff
@@ -31,31 +34,36 @@ def test_cplerror_str():
     assert str(err) == "test123"
 
 
+@pytest.mark.xfail(gdal_version.at_least('3.3'), reason="GDAL <3.3 will not warn")
```

Comment:
It seems that only GDAL >=3.3 will raise error message (I doesn't seems to be linked with PROJ version, I tested GDAL3.3.3+PROJ 7 locally, and test pass for GDAL3.3/3.4 + PROJ 8 in GitHub actions)